### PR TITLE
Clean up and add more tests for support for multiple UUIDs

### PR DESCRIPTION
**Pull request by nileema**

### CODE REVIEW

**Comment by nileema** [comment]
I'll squash these two commits before merging.

File: presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/ShardPredicate.java [hunk 1]
```diff
@@ -206,10 +156,40 @@ public static ShardPredicate create(TupleDomain<RaptorColumnHandle> tupleDomain)
...
         }
-
         return new ShardPredicate(predicate.toString(), types.build(), values.build());
     }
 
+    private static String createShardPredicate(ImmutableList.Builder<JDBCType> types, ImmutableList.Builder<Object> values, Domain domain, JDBCType jdbcType)
+    {
+        List<Range> ranges = domain.getValues().getRanges().getOrderedRanges();
+
+        // if we do not have single values to match against, don't apply a special predicate
```
Comment (electrum) [inline presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/ShardPredicate.java]: How about "only apply predicates if all ranges are single values"

File: presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/ShardPredicate.java [hunk 2]
```diff
@@ -206,10 +156,40 @@ public static ShardPredicate create(TupleDomain<RaptorColumnHandle> tupleDomain)
...
+        if (!ranges.stream().allMatch(Range::isSingleValue)) {
```
Comment (electrum) [inline presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/ShardPredicate.java]: Maybe make this
```java
if (ranges.isEmpty() || !ranges.stream().allMatch(Range::isSingleValue)) {
```
Then remove the `.setEmptyValue()`

File: presto-raptor/src/test/java/com/facebook/presto/raptor/metadata/TestShardPredicate.java [hunk 3]
```diff
@@ -40,14 +42,14 @@ public void testSimpleShardUuidPredicate()
             throws Exception
     {
         String uuid = randomUUID().toString();
-        TupleDomain<RaptorColumnHandle> tupleDomain = TupleDomain.withColumnDomains(ImmutableMap.of(
-                shardUuidColumnHandle("test"), Domain.singleValue(VARCHAR, utf8Slice(uuid))
+        TupleDomain<RaptorColumnHandle> tupleDomain = withColumnDomains(ImmutableMap.of(
+                shardUuidColumnHandle("test"), singleValue(VARCHAR, utf8Slice(uuid))
         ));
```
Comment (electrum) [inline presto-raptor/src/test/java/com/facebook/presto/raptor/metadata/TestShardPredicate.java]: Put this on previous line

**Comment by nileema** [comment]
Merged.

**Pull request closed without merging**